pace-util/pace/util/report.py

```diff
@@ -0,0 +1,114 @@
+import copy
+import json
+from datetime import datetime
+from typing import Any, Dict, List
+
+import numpy as np
+
+from pace.util.mpi import MPI
+
+
+def set_experiment_info(
+    experiment_name: str, time_step: int, backend: str, git_hash: str
+) -> Dict[str, Any]:
+    experiment: Dict[str, Any] = {}
+    now = datetime.now()
+    dt_string = now.strftime("%d/%m/%Y %H:%M:%S")
+    experiment["setup"] = {}
+    experiment["setup"]["timestamp"] = dt_string
+    experiment["setup"]["dataset"] = experiment_name
+    experiment["setup"]["timesteps"] = time_step
+    experiment["setup"]["hash"] = git_hash
+    experiment["setup"]["version"] = "python/" + backend
+    experiment["setup"]["format_version"] = 3
+    experiment["times"] = {}
+    return experiment
+
+
+def collect_keys_from_data(times_per_step: List[Dict[str, float]]) -> List[str]:
+    """Collects all the keys in the list of dics and returns a sorted version"""
+    keys = set()
+    for data_point in times_per_step:
+        for k, _ in data_point.items():
+            keys.add(k)
+    sorted_keys = list(keys)
+    sorted_keys.sort()
+    return sorted_keys
+
+
+def gather_timing_data(
+    times_per_step: List[Dict[str, float]],
+    results: Dict[str, Any],
+    comm: MPI.Comm,
```

Comment:
This type hint will cause an exception if mpi4py (an optional dependency) is not installed, in which case MPI will be None.

Reply:
Removed type hint here.